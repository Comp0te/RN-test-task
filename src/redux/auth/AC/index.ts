import { createAction, ActionsUnion } from '../../../shared/helpers/createAction';

export enum ActionTypes {
  AUTH_LOGOUT = 'AUTH_LOGOUT',
  AUTH_SET_IS_AUTH_USER = 'AUTH_SET_IS_AUTH_USER',
}

export const Actions = {
  logout: () => createAction(ActionTypes.AUTH_LOGOUT),
  setIsAuthUser: (data: boolean) => createAction(ActionTypes.AUTH_SET_IS_AUTH_USER, {data}),
};

export type Actions = ActionsUnion<typeof Actions>;
