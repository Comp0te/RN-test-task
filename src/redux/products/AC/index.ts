import { createAction, ActionsUnion } from '../../../shared/helpers/createAction';
import { ProductModel } from '../../../shared/models/product.model';

export enum ActionTypes {
  SET_PRODUCTS_DATA = 'SET_PRODUCTS_DATA',
}

export const Actions = {
  setProductsData: (data: ProductModel[]) => createAction(ActionTypes.SET_PRODUCTS_DATA, {data}),
};

export type Actions = ActionsUnion<typeof Actions>;
