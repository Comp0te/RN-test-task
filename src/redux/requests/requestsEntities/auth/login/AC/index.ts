import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { AuthInput } from '../../../../../../shared/services/auth.service';
import { AuthResponse } from '../../../../types';

export enum ActionTypes {
  LOGIN_REQUEST = 'LOGIN_REQUEST',
  LOGIN_REQUEST_SUCCESS = 'LOGIN_REQUEST_SUCCESS',
  LOGIN_REQUEST_FAIL = 'LOGIN_REQUEST_FAIL',
}

export const Actions = {
  login: (data: AuthInput) => createAction(ActionTypes.LOGIN_REQUEST, {data}),
  loginSuccess: (data: AuthResponse) => {
    return createAction(ActionTypes.LOGIN_REQUEST_SUCCESS, {data});
  },
  loginFail: (errors: string) => {
    return createAction(ActionTypes.LOGIN_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
