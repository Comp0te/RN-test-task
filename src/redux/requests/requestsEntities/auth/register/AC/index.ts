import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { AuthInput } from '../../../../../../shared/services/auth.service';
import { AuthResponse } from '../../../../types';

export enum ActionTypes {
  REGISTER_REQUEST = 'REGISTER_REQUEST',
  REGISTER_REQUEST_SUCCESS = 'REGISTER_REQUEST_SUCCESS',
  REGISTER_REQUEST_FAIL = 'REGISTER_REQUEST_FAIL',
}

export const Actions = {
  register: (data: AuthInput) => createAction(ActionTypes.REGISTER_REQUEST, {data}),
  registerSuccess: (data: AuthResponse) => {
    return createAction(ActionTypes.REGISTER_REQUEST_SUCCESS, {data});
  },
  registerFail: (errors: string) => {
    return createAction(ActionTypes.REGISTER_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
