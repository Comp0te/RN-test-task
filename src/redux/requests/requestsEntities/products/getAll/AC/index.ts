import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { ProductModel } from '../../../../../../shared/models/product.model';

export enum ActionTypes {
  PRODUCTS_GET_ALL_REQUEST = 'PRODUCTS_GET_ALL_REQUEST',
  PRODUCTS_GET_ALL_REQUEST_SUCCESS = 'PRODUCTS_GET_ALL_REQUEST_SUCCESS',
  PRODUCTS_GET_ALL_REQUEST_FAIL = 'PRODUCTS_GET_ALL_REQUEST_FAIL',
}

export const Actions = {
  getAllProducts: () => createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST),
  getAllProductsSuccess: (data: ProductModel[]) => {
    return createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST_SUCCESS, {data});
  },
  getAllProductsFail: (errors: string) => {
    return createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
