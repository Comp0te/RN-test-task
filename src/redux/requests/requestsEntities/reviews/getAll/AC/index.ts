import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { ReviewResponse } from '../../../../../../shared/models/review.model';

export enum ActionTypes {
  REVIEWS_GET_ALL_REQUEST = 'REVIEWS_GET_ALL_REQUEST',
  REVIEWS_GET_ALL_REQUEST_SUCCESS = 'REVIEWS_GET_ALL_REQUEST_SUCCESS',
  REVIEWS_GET_ALL_REQUEST_FAIL = 'REVIEWS_GET_ALL_REQUEST_FAIL',
}

export const Actions = {
  getAllReviews: (data: string) => createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST, {data}),
  getAllReviewsSuccess: (data: ReviewResponse[]) => {
    return createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST_SUCCESS, {data});
  },
  getAllReviewsFail: (errors: string) => {
    return createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
