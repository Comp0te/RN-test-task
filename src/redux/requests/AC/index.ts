import * as login from '../requestsEntities/auth/login/AC';
import * as register from '../requestsEntities/auth/register/AC';
import * as getAllProducts from '../requestsEntities/products/getAll/AC';
import * as getAllReviews from '../requestsEntities/reviews/getAll/AC';
import * as postReview from '../requestsEntities/reviews/post/AC';

export const requestsAC = {
  login,
  register,
  getAllProducts,
  getAllReviews,
  postReview,
};