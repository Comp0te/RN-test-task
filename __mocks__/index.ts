import { AuthInput } from '../src/shared/services/auth.service';
import { AuthResponse } from '../src/redux/requests/types';
import { ProductModel } from '../src/shared/models/product.model';
import { ReviewResponse } from '../src/shared/models/review.model';
import { IReviewPostInput } from '../src/shared/services/reviews.service';

export const authInputMock: AuthInput = {
  username: 'Compote',
  password: 'password',
};

export const authResponseMock: AuthResponse = {
  success: true,
  token: 'token',
};

export const authResponseFailMock = {
  success: false,
  message: 'Ooops',
};

export const productMock: ProductModel = {
  id: 1,
  img: 'img',
  text: 'text',
  title: 'title',
};

export const reviewResponseMock: ReviewResponse = {
  id: 1,
  created_by: {
    id: 1,
    username: 'bla-bla',
  },
  product: 1,
  text: 'text',
  rate: 5,
};

export const reviewPostInputMock: IReviewPostInput = {
  text: 'text',
  rate: 2,
  product_id: 'id',
};