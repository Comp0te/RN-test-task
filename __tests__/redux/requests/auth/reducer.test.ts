import { requestsAC } from '../../../../src/redux/requests/AC';
import {
  reducer as loginReducer,
  initialState as loginInitialState,
} from '../../../../src/redux/requests/requestsEntities/auth/login/reducers';
import {
  reducer as registerReducer,
  initialState as registerInitialState,
} from '../../../../src/redux/requests/requestsEntities/auth/register/reducers';
import { authInputMock, authResponseMock } from '../../../../__mocks__';

describe('Redux auth request reducers', () => {
  it('update state after LOGIN_REQUEST action', () => {
    expect(loginReducer(
      loginInitialState,
      requestsAC.login.Actions.login(authInputMock)),
    ).toMatchSnapshot();
  });

  it('update state after LOGIN_REQUEST_SUCCESS action', () => {
    expect(loginReducer(
      loginInitialState,
      requestsAC.login.Actions.loginSuccess(authResponseMock)),
    ).toMatchSnapshot();
  });

  it('update state after LOGIN_REQUEST_FAIL action', () => {
    expect(loginReducer(
      loginInitialState,
      requestsAC.login.Actions.loginFail('error')),
    ).toMatchSnapshot();
  });

  it('update state after REGISTER_REQUEST action', () => {
    expect(registerReducer(
      registerInitialState,
      requestsAC.register.Actions.register(authInputMock)),
    ).toMatchSnapshot();
  });

  it('update state after REGISTER_REQUEST_SUCCESS action', () => {
    expect(registerReducer(
      registerInitialState,
      requestsAC.register.Actions.registerSuccess(authResponseMock)),
    ).toMatchSnapshot();
  });

  it('update state after REGISTER_REQUEST_FAIL action', () => {
    expect(registerReducer(
      registerInitialState,
      requestsAC.register.Actions.registerFail('error')),
    ).toMatchSnapshot();
  });
});
