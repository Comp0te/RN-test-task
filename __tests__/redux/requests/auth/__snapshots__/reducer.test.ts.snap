// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux auth request reducers update state after LOGIN_REQUEST action 1`] = `
Object {
  "data": null,
  "errors": null,
  "isLoading": true,
}
`;

exports[`Redux auth request reducers update state after LOGIN_REQUEST_FAIL action 1`] = `
Object {
  "data": null,
  "errors": "error",
  "isLoading": false,
}
`;

exports[`Redux auth request reducers update state after LOGIN_REQUEST_SUCCESS action 1`] = `
Object {
  "data": Object {
    "success": true,
    "token": "token",
  },
  "errors": null,
  "isLoading": false,
}
`;

exports[`Redux auth request reducers update state after REGISTER_REQUEST action 1`] = `
Object {
  "data": null,
  "errors": null,
  "isLoading": true,
}
`;

exports[`Redux auth request reducers update state after REGISTER_REQUEST_FAIL action 1`] = `
Object {
  "data": null,
  "errors": "error",
  "isLoading": false,
}
`;

exports[`Redux auth request reducers update state after REGISTER_REQUEST_SUCCESS action 1`] = `
Object {
  "data": Object {
    "success": true,
    "token": "token",
  },
  "errors": null,
  "isLoading": false,
}
`;
